skylog/storage/types.go

```diff
@@ -0,0 +1,9 @@
+package storage
+
+import "github.com/google/trillian/merkle/compact"
+
+// Node represents a Merkle tree node.
+type Node struct {
```

Comment:
Might want to think about making `Node` immutable, though I don't think the `compact.NodeID` is so might not help that much.

Reply:
Could you elaborate what you mean by immutable? Like unexporting the fields and adding read-only getters?

Reply:
Cannot be modified after construction. Making IDs opaque outside the defining package is another option, but not as easy in Go where it's all or nothing visibility. 

In `storage.NodeID` we ended up with a performance penalty copying byte slices because there were mutator methods directly on the IDs. That doesn't apply here though so the decision is really about whether it has benefits for correctness. It's also possibly too early days to say so maybe carry on like this for the moment until more of it takes shape.